elephant/statistics.py

```diff
@@ -240,3 +243,273 @@ def lv(v):
+        filter function is aligned with the spike to convolve
+        Default:None
+    m_idx : int
+        index of the value in the kernel function vector that corresponds
+        to its gravity center. this parameter is not mandatory for
+        symmetrical kernels but it is required when asymmetrical kernels
+        are to be aligned at their gravity center with the event times if None
+        is assumed to be the median value of the kernel support
+        Default : None
+    trim : bool
+        if True, only the 'valid' region of the convolved
+        signal are returned, i.e., the points where there
+        isn't complete overlap between kernel and spike train
+        are discarded
+        NOTE: if True and an asymmetrical kernel is provided
+        the output will not be aligned with [t_start, t_stop]
+
+    Returns
+    -------
+    rate : neo.AnalogSingal
+        Contains the rate estimation in unit Hertz (Hz).
+        Has a property 'rate.times' which contains the time axis of the rate
+        estimate. The unit of this property is the same as the resolution that
+        is given as an argument to the function.
+
+    See also:
+        statistics.make_kernel
+    """
+    kernel, norm, m_idx = make_kernel(form=form, sigma=sigma, resolution=resolution)
+    units = pq.CompoundUnit("%s*s" % str(resolution.rescale('s').magnitude))
+    spiketrain = spiketrain.rescale(units)
+    if t_start is None:
+        t_start = spiketrain.t_start
+    else:
+        t_start = t_start.rescale(spiketrain.units)
+
+    if t_stop is None:
+        t_stop = spiketrain.t_stop
+    else:
+        t_stop = t_stop.rescale(spiketrain.units)
+
+    if m_idx is None:
+        m_idx = kernel.size / 2
+
+    time_vector = np.zeros(int((t_stop - t_start)) + 1)
+
+    spikes_slice = spiketrain.time_slice(t_start, t_stop) \
+        if len(spiketrain) else np.array([])
+
+    for spike in spikes_slice:
+        index = int((spike - t_start))
+        time_vector[index] += 1
+
+    r = norm * scipy.signal.fftconvolve(time_vector, kernel, 'full')
+    if np.any(r < 0):
+        warnings.warn('Instantaneous firing rate approximation contains'
+                      'negative values, possibly caused due to machine'
+                      'precision errors')
+
+    if acausal:
+        if not trim:
+            r = r[m_idx:-(kernel.size - m_idx)]
+
+        elif trim:
+            r = r[2 * m_idx:-2*(kernel.size - m_idx)]
+            t_start = t_start + m_idx * spiketrain.units
+            t_stop = t_stop - ((kernel.size) - m_idx) * spiketrain.units
+
+    else:
+        if not trim:
+            r = r[m_idx:-(kernel.size - m_idx)]
+
+        elif trim:
+            r = r[2 * m_idx:-2*(kernel.size - m_idx)]
+            t_start = t_start + m_idx * spiketrain.units
+            t_stop = t_stop - ((kernel.size) - m_idx) * spiketrain.units
+
+    rate = neo.AnalogSignal(signal=r, sampling_period=resolution, units=pq.Hz,
```

Comment:
`AnalogSignal` will be removed in the next version of Neo, so this should use `AnalogSignalArray`.